src/app/pages/TransactionDetailPage/index.tsx

```diff
@@ -174,6 +178,33 @@ const TransactionInfoTooltip: FC<PropsWithChildren<{ label: string }>> = ({ labe
   )
 }
 
+export const ShowHideData: FC<{ data: string; threshold: number }> = ({ data, threshold }) => {
+  const { t } = useTranslation()
+  const [showData, setShowData] = useState(false)
+  const needsHiding = data.length > threshold
+  if (!needsHiding) {
+    return (
+      <Typography variant={'mono'} sx={{ overflowWrap: 'anywhere' }}>
+        {data}
+      </Typography>
+    )
+  }
+  return (
+    <div>
+      <Collapse orientation={'vertical'} in={showData} collapsedSize={'3em'}>
+        <Typography variant={'mono'} sx={{ overflowWrap: 'anywhere' }}>
+          {data}
+        </Typography>
+      </Collapse>
+      {data.length! > threshold && (
```

Comment:
what's the purpose of `!` in `data.length!` ?

Reply:
Nothing, it was a typo, fixed.